src/controllers/dynakube/oneagent/daemonset/labels.go

```diff
@@ -2,9 +2,14 @@ package daemonset
 
 import "github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
 
-// buildLabels returns generic labels based on the name given for a Dynatrace OneAgent
 func BuildLabels(name string, feature string) map[string]string {
 	labels := kubeobjects.CommonLabels(name, kubeobjects.OneAgentComponentLabel)
 	labels[kubeobjects.FeatureLabel] = feature
 	return labels
 }
+
+func buildMatchLabels(name string) map[string]string {
+	labels := kubeobjects.CommonLabels(name, kubeobjects.OneAgentComponentLabel)
+	delete(labels, kubeobjects.AppVersionLabel)
```

Comment:
how does that make sense?
that would mean if we upgrade the operator outdated pods would match?

Reply:
its what we had before,
we never had the version or anything interesting for matchLabels,
and its not important, the operator would update the will the daemonset because the labels changed due to the version, the matchLabels not changing is fine

we could just always delete the previous daemonset, but then there would be a "not nice" period if while both daemonsets exists(1 terminating 1 crashlooping then running)
also I don't know how this would work in case of readonly (we would want to mount the same thing but it might be still used)